internal/provider/kubernetes/controller.go

```diff
@@ -1487,16 +1493,30 @@ func (r *gatewayAPIReconciler) processGateways(ctx context.Context, managedGC *g
 	return nil
 }
 
-func (r *gatewayAPIReconciler) processServiceCluster(resourceName string, resourceMap *resourceMappings) {
+func (r *gatewayAPIReconciler) processServiceCluster(ep *egv1a1.EnvoyProxy, resourceNamespacedName types.NamespacedName, resourceMap *resourceMappings) {
```

Comment:
not a block, we should add a test somewhere to make sure that this will not be broken again.